packages/web/src/services/competitionService.ts

```diff
@@ -0,0 +1,98 @@
+// services/competitionService.ts
+import { useState, useEffect } from "react";
+import { supabase } from "./supabaseClient";
+
+export interface Competition {
+  id: number;
+  name: string;
+  created_by: string;
+  created_at: string;
+}
+
+export function useCompetition() {
+  const [competition, setCompetition] = useState<Competition | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<Error | null>(null);
+
+  const fetchCompetition = async () => {
+    try {
+      setIsLoading(true);
+      const {
+        data: { user },
+      } = await supabase.auth.getUser();
+
+      if (!user) throw new Error("No authenticated user found");
+
+      const { data, error } = await supabase
+        .from("competitions")
+        .select("*")
+        .eq("created_by", user.id)
+        .single();
+
+      if (error && error.code !== "PGRST116") {
+        // PGRST116 is the "no rows returned" error
+        throw error;
+      }
+
+      setCompetition(data);
+    } catch (err) {
+      setError(err instanceof Error ? err : new Error("An error occurred"));
```

Comment:
If you decide to NOT reset the value of `competition` as suggested above, you absolutely should have it reset here. The state of having a successful output and an error should never coexist.